service/internal/builder/pipelines_builder.go

```diff
@@ -52,7 +53,7 @@ type BuiltPipelines map[config.ComponentID]*builtPipeline
 func (bps BuiltPipelines) StartProcessors(ctx context.Context, host component.Host) error {
 	for _, bp := range bps {
 		bp.logger.Info("Pipeline is starting...")
-		hostWrapper := components.NewHostWrapper(host, bp.logger)
+		hostWrapper := components.NewHostWrapper(host, bp.componentID, bp.logger)
```

Comment:
I see. I will have to have host wrapper per processor then, and find a way to easily access the component id. Speaking of which, is there a reason why the component id is not part of the component interface?

Reply:
Yes, it is for the moment used in the config.Receiver interface, so to avoid circular dependencies needs to be there. I have a proposal to change that.

Reply:
I could not find a way to access to the component id of the processor in `StartProcessors`, so I added a map to `builtPipeline` to lookup the component id by processor. I don't love this solution, but I couldn't find anything I liked better.